pkg/dbmate/db.go

```diff
@@ -453,6 +463,32 @@ func (db *DB) Rollback() error {
 	return nil
 }
 
+func waitForOthersChangeLockToReleaseAndAcquireChangeLock(drv Driver, sqlDB *sql.DB) error {
+	counter := 0
+	for { // FIXME should we have a timeout for change locks? Or no?
```

Comment:
rather than manually implementing the loop, could you instead just use the non-try versions of the locking functions? e.g. `pg_advisory_lock` instead of `pg_try_advisory_lock`

that way it would just automatically block until the lock is available. and it would save you manually worrying about the timeout.

Reply:
Sure! This was solely for the the ticking `...` effect, which in a way I like. For me it would make it easier to understand how long the wait has been, if something is long-running.